Framework/Core/include/Framework/DataChunk.h

```diff
@@ -10,17 +10,15 @@
 #ifndef FRAMEWORK_DATACHUNK_H
 #define FRAMEWORK_DATACHUNK_H
 
+#include "MemoryResources/MemoryResources.h"
+
 namespace o2
 {
 namespace framework
 {
-
-/// Simple struct to hold a pointer to the actual FairMQMessage.
-/// In principle this could be an iovec...
-struct DataChunk {
-  char *data;
-  size_t size;
-};
+// FIXME: make sure that a DataChunk can not be copied or assigned, because the context returns the
+// object by reference and we have to make sure that the code is using the reference instead of a copy
+using DataChunk = o2::vector<char>;
```

Comment:
Why using `o2::vector<char>` here? Can't we simply use `std::pmr::vector<char>`? This way we avoid the namespace issues we had elsewhere (i.e. `using namespace std;` introduced  ambiguity).
The idea of o2::vector<T>` was to reduce a bit of typing if users were manipulating / creating it directly, however since this is only for `T=char` and it's in any case aliased to `DataChunk`, I think we can use directly the full type.

Reply:
I have to check, the `o2::pmr` stuff is using boost, and I have not yet checked the compatibility and how the `polymorphic_allocator` implementations can be interchanged. On the long run this will probably be converted to `std::pmr`

Reply:
My worry is that this will break any code which does `using namespace std;` and includes `std::vector`.

Reply:
and I thought we aim to expose `o2::vector` to the user in general

Reply:
We aim to have a polymorphically allocated backend for our data, so that we can resize messages (which we cannot do right now). I think the issue with `using` which we had already shows that we do not want to use an alias for `o2::vector`. For the same reason I would keep the `span` maker.

Reply:
ok, so the first step towards this is the support for resizable `DataChunk` which is working now and unit test is added.